drivers/usb/common/stm32/stm32_usb_pwr.c

```diff
@@ -118,3 +119,62 @@ int stm32_usb_pwr_enable(void)
 
 	return err;
 }
+
+int stm32_usb_pwr_disable(void)
+{
+	atomic_val_t old = atomic_dec(&usb_pwr_refcount);
```

Comment:
Are atomic operations sufficient here. Consider 2 threads running
- Thread A runs and has enabled a USB device;
- Thread A starts to shutdown its USB device, calls `stm32_usb_pwr_disable()` that returns with `old == 0`;
- Thread B gets scheduled, powers up another USB device: calls `stm32_usb_pwr_enable()` and start enabling already enabled USB common power supplies;
- Thread A gets scheduled and completes its `stm32_usb_pwr_disable()` sequence.

Don't you need a mutex or like here?